src/drivers/unified_gamepad/capability.rs

```diff
@@ -0,0 +1,707 @@
+     * e.g. G1-G18, or S1 - S30. The KEY_MACRO# codes MUST NOT be used for keys
+     * where the marking on the key does indicate a defined meaning / purpose.
+     *
+     * The KEY_MACRO# codes MUST also NOT be used as fallback for when no existing
+     * KEY_FOO define matches the marking / purpose. In this case a new KEY_FOO
+     * define MUST be added.
+     */
+    KeyboardKeyMacro1 = 0x290,
+    KeyboardKeyMacro2 = 0x291,
+    KeyboardKeyMacro3 = 0x292,
+    KeyboardKeyMacro4 = 0x293,
+    KeyboardKeyMacro5 = 0x294,
+    KeyboardKeyMacro6 = 0x295,
+    KeyboardKeyMacro7 = 0x296,
+    KeyboardKeyMacro8 = 0x297,
+    KeyboardKeyMacro9 = 0x298,
+    KeyboardKeyMacro10 = 0x299,
+    KeyboardKeyMacro11 = 0x29a,
+    KeyboardKeyMacro12 = 0x29b,
+    KeyboardKeyMacro13 = 0x29c,
+    KeyboardKeyMacro14 = 0x29d,
+    KeyboardKeyMacro15 = 0x29e,
+    KeyboardKeyMacro16 = 0x29f,
+    KeyboardKeyMacro17 = 0x2a0,
+    KeyboardKeyMacro18 = 0x2a1,
+    KeyboardKeyMacro19 = 0x2a2,
+    KeyboardKeyMacro20 = 0x2a3,
+    KeyboardKeyMacro21 = 0x2a4,
+    KeyboardKeyMacro22 = 0x2a5,
+    KeyboardKeyMacro23 = 0x2a6,
+    KeyboardKeyMacro24 = 0x2a7,
+    KeyboardKeyMacro25 = 0x2a8,
+    KeyboardKeyMacro26 = 0x2a9,
+    KeyboardKeyMacro27 = 0x2aa,
+    KeyboardKeyMacro28 = 0x2ab,
+    KeyboardKeyMacro29 = 0x2ac,
+    KeyboardKeyMacro30 = 0x2ad,
+
+    /*
+     * Some keyboards with the macro-keys described above have some extra keys
+     * for controlling the host-side software responsible for the macro handling:
+     * -A macro recording start/stop key. Note that not all keyboards which emit
+     *  KEY_MACRO_RECORD_START will also emit KEY_MACRO_RECORD_STOP if
+     *  KEY_MACRO_RECORD_STOP is not advertised, then KEY_MACRO_RECORD_START
+     *  should be interpreted as a recording start/stop toggle;
+     * -Keys for switching between different macro (pre)sets, either a key for
+     *  cycling through the configured presets or keys to directly select a preset.
+     */
+    KeyboardKeyMacroRecordStart = 0x2b0,
+    KeyboardKeyMacroRecordStop = 0x2b1,
+    KeyboardKeyMacroPresetCycle = 0x2b2,
+    KeyboardKeyMacroPreset1 = 0x2b3,
+    KeyboardKeyMacroPreset2 = 0x2b4,
+    KeyboardKeyMacroPreset3 = 0x2b5,
+
+    /*
+     * Some keyboards have a buildin LCD panel where the contents are controlled
+     * by the host. Often these have a number of keys directly below the LCD
+     * intended for controlling a menu shown on the LCD. These keys often don't
+     * have any labeling so we just name them KEY_KBD_LCD_MENU#
+     */
+    KeyboardKeyKbdLcdMenu1 = 0x2b8,
+    KeyboardKeyKbdLcdMenu2 = 0x2b9,
+    KeyboardKeyKbdLcdMenu3 = 0x2ba,
+    KeyboardKeyKbdLcdMenu4 = 0x2bb,
+    KeyboardKeyKbdLcdMenu5 = 0x2bc,
+
+    MouseButtonLeft = 0x110,
+    MouseButtonRight = 0x111,
+    MouseButtonMiddle = 0x112,
+    MouseButtonSide = 0x113,
+    MouseButtonExtra = 0x114,
+
+    /// Base button, usually on the bottom right, Steam Quick Access Button (...)
```

Comment:
We should avoid calling this Base Button, those buttons are reserved for the physical base of a joystick. Only the openSD driver uses BTN_BASE for this, which is wrong.